sirepo/pkcli/test_http.py

```diff
@@ -5,21 +5,32 @@
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
 from __future__ import absolute_import, division, print_function
+from pykern import pkcompat
 from pykern import pkconfig
+from pykern import pkinspect
 from pykern import pkjson
 from pykern.pkcollections import PKDict
-from pykern.pkdebug import pkdp, pkdexc, pkdlog
+from pykern.pkdebug import pkdp, pkdexc, pkdlog, pkdformat
 import asyncio
 import contextlib
 import copy
+import inspect
 import re
 import sirepo.sim_data
 import sirepo.util
 import time
 import tornado.httpclient
+import tornado.ioloop
+import tornado.locks
 import random
+import signal
+import sys
 
-CODES = PKDict(
+cfg = None
+
+_sims = []
+
+_CODES = PKDict(
```

Comment:
A tuple can't be copy()'ed or shuffle()'ed which I need to do on these lists. I could use deepcopy() instead of copy() and cast to a list then back to a tuple but that doesn't seem worth it. Agree?

Reply:
The tuple should be deepcopied. When you need to shuffle, you can use `sample(x, k=len(x))`